ReactWindows/ReactNative/Views/Text/ReactTextViewManager.cs

```diff
@@ -185,5 +187,203 @@ protected override RichTextBlock CreateViewInstance(ThemedReactContext reactCont
+        }
+
+        /// <summary>
+        /// Called when view is detached from view hierarchy and allows for 
+        /// additional cleanup by the <see cref="ReactTextViewManager"/>.
+        /// </summary>
+        /// <param name="reactContext">The React context.</param>
+        /// <param name="view">The view.</param>
+        public override void OnDropViewInstance(ThemedReactContext reactContext, RichTextBlock view)
+        {
+            base.OnDropViewInstance(reactContext, view);
+            view.SelectionChanged -= OnSelectionChanged;
+        }
+
+        /// <summary>
+        /// Called when selection changes for the RichTextBlock.
+        /// </summary>
+        /// <param name="sender">The RichTextBlock that is the source of the event.</param>
+        /// <param name="e">Event arguments.</param>
+        private void OnSelectionChanged(object sender, RoutedEventArgs e)
+        {
+            RichTextBlock view = sender as RichTextBlock;
+            string selection = GetSelectedString(view);
+
+            // Fire the event.
+            view.GetReactContext()
+                .GetNativeModule<UIManagerModule>()
+                .EventDispatcher
+                .DispatchEvent(
+                    new ReactTextBlockSelectionEvent(
+                        view.GetTag(),
+                        selection));
+        }
+
+        /// <summary>
+        /// Returns a text representation of the selection in the RichTextBlock.
+        /// </summary>
+        /// <param name="view">The RichTextBlock.</param>
+        /// <returns>Text representation of the selection.</returns>
+        private string GetSelectedString(RichTextBlock view)
+        {
+            TextPointer selectionStart = view.SelectionStart;
+            TextPointer selectionEnd = view.SelectionEnd;
+
+            return GetStringByStartAndEndPointers(view, selectionStart, selectionEnd);
+        }
+
+        /// <summary>
+        /// Returns a text representation of the elements between <paramref name="start"/> and <paramref name="end"/> in the RichTextBlock.
+        /// </summary>
+        /// <param name="view">The RichTextBlock.</param>
+        /// <param name="start">Start TextPointer.</param>
+        /// <param name="end">End TextPointer.</param>
+        /// <returns></returns>
+        public static string GetStringByStartAndEndPointers(RichTextBlock view, TextPointer start, TextPointer end)
+        {
+            if (start == null || end == null || start == end)
+            {
+                return null;
+            }
+
+            // Parse the RichTextBlock contents.
+            StringBuilder selectedText = new StringBuilder();
+            Debug.Assert(view.Blocks.Count == 1, "RichTextBlock is expected to contain only one paragraph.");
+            Paragraph paragraph = view.Blocks.First() as Paragraph;
+            ProcessSelectedInlines(paragraph.Inlines, selectedText, false, start, end);
```

Comment:
You use `as` to cast to `Paragraph`, but don't have a null check. Either use convert expression or null check.

Reply:
Fixed.

---
In reply to: [215225002](https://github.com/Microsoft/react-native-windows/pull/1951#discussion_r215225002) [](ancestors = 215225002)